volplugin/volplugin.go

```diff
@@ -95,3 +110,45 @@ func logHandler(name string, debug bool, actionFunc func(http.ResponseWriter, *h
 		actionFunc(w, r)
 	}
 }
+
+func (dc *DaemonConfig) updateMounts() error {
+	cd := ceph.NewDriver()
+	mounts, err := cd.Mounted()
+	if err != nil {
+		return err
+	}
+
+	for _, mount := range mounts {
+		parts := strings.Split(mount.Volume.Name, "/")
+		if len(parts) != 2 {
+			log.Warnf("Invalid volume named %q in ceph scan: skipping refresh", mount.Volume.Name)
+			continue
+		}
+
+		log.Infof("Refreshing existing mount for %q", mount.Volume.Name)
+
+		volConfig, err := requestVolumeConfig(dc.Master, parts[0], parts[1])
+		if err != nil {
```

Comment:
I am trying to understand the conditions when the err will be not nil here, in order to verify the possible handling. Let me know if I got it right.

One of the conditions when this can happen is when a volume was deleted (from volmaster) while volplugin was down. In that case we might just want to cleanup the mount and move on.

Another would be when volmaster itself is unreachable. In that case we might not be able to do much but just return error and wait for instructions from volmaster when it comes up like it can reply whatever it has.

Is there any other error condition that we need to think about?